modules/mask.py

```diff
@@ -1,153 +1,95 @@
 import numpy as np
-import numpy.ma as ma
-import xarray as xr
-
-def load_land_mask(data_path):
-    """Loads land fraction file from
-    http://climate-modelling.canada.ca/climatemodeldata/canrcm/CanRCM4/NAM-44_ECMWF-ERAINT_evaluation/fx/atmos/sftlf/index.shtml
-    and creates a mask for cells that are 0% land.
-    ----------------------------------------------
-    Args:
-        data_path (str): path to the land fraction mask file
-    Returns:
-        mask (numpy.masked_array): masked array based on file
-    """
-    ds_mask = xr.open_dataset(data_path)
-    mask = ma.masked_greater(ds_mask['sftlf'].values, 10.).mask
-    return mask
-
-def mask_land_and_nan(dv_field, mask_land):
-    """Creates a master mask by masking any cell in the ensemble
-    that has a NaN value. If a NaN value is found in the ensemble,
-    that grid cell is masked for every ensemble member, thus
-    disqualifying that grid cell from further analysis. It is then
-    combined with the land mask, to create a master mask of cells
-    to use in the analysis.
-    ------------------------------------------------------------
-    Args:
-        dv_field (numpy.ndarray): datacube containing the ensemble members
-        mask_land (numpy.masked_array): land mask created by load_land_mask()
-
-    Returns:
-        mask_master (numpy.ndarray): a CanRCM4 field shaped array containing
-            a boolean master mask
-    """
-    nan_mask = np.apply_over_axes(
-                    np.logical_or.reduce,
-                    np.isnan(dv_field),
-                    (0)
-    )
-
-    mask_master = ~np.logical_or(~mask_land, nan_mask)
-
-    return mask_master
-
-def flatten_mask(mask_master):
-    """Reshapes the master mask created in mask_master into
-    the flattened ensemble shape, that is, (number of grid cells) x (number of
-    ensemble members). Each row in this new shape represents a grid cell
-    from the ensemble members.
-
+from shapely.geometry import Point
+import geopandas as gpd
+
+
+def check_polygon(p):
+    if not isinstance(p, gpd.GeoSeries):
+        raise TypeError('Must be gpd.GeoSeries, not {}'.format(type(p)))
+    if p.hasnans:
+        raise ValueError('GeoSeries contains invalid values.')
+    if not p.size > 0:
+        raise ValueError('GeoSeries is is_empty')
+    return True
+
+
+def check_pre_proj(p):
```

Comment:
These two functions seem like a strange way to validate a polygon. And... it only works for a polygon representing Canada in one specific projection? Can you explain more about the utility of this?

Reply:
I want to verify that the polygon provided is in the regular projection. Is that an unnecessary requirement? I used area as a preliminary way to test this, but agree that it should be more general. 

I've since found that `geopandas` has a `crs` attribute where I can check the projection directly.